src/Compatibility/Microsoft.DotNet.Compatibility/ValidatePackage.cs

```diff
@@ -96,45 +90,40 @@ protected override void ExecuteCore()
-            CompatibilityLogger logger = new(Log, CompatibilitySuppressionFilePath, GenerateCompatibilitySuppressionFile, NoWarn);
-
-            new CompatibleTfmValidator(RunApiCompat, EnableStrictModeForCompatibleTfms, logger, apiCompatReferences).Validate(package);
-            new CompatibleFrameworkInPackageValidator(EnableStrictModeForCompatibleFrameworksInPackage, logger, apiCompatReferences).Validate(package);
-
-            if (!DisablePackageBaselineValidation && !string.IsNullOrEmpty(BaselinePackageTargetPath))
-            {
-                Package baselinePackage = NupkgParser.CreatePackage(BaselinePackageTargetPath, runtimeGraph);
-                new BaselinePackageValidator(baselinePackage, RunApiCompat, logger, apiCompatReferences).Validate(package);
-            }
-
-            if (GenerateCompatibilitySuppressionFile)
-            {
-                logger.GenerateSuppressionsFile(CompatibilitySuppressionFilePath);
-            }
+            MSBuildCompatibilityLogger log = new(Log, CompatibilitySuppressionFilePath, GenerateCompatibilitySuppressionFile, NoWarn);
+            ValidatorBootstrap.RunValidators(log,
+                PackageTargetPath!,
+                GenerateCompatibilitySuppressionFile,
+                RunApiCompat,
+                EnableStrictModeForCompatibleTfms,
+                EnableStrictModeForCompatibleFrameworksInPackage,
+                DisablePackageBaselineValidation,
+                BaselinePackageTargetPath,
+                RuntimeGraph,
+                apiCompatReferences);
         }
 
 #if NETCOREAPP
-        private Assembly ResolverForRoslyn(AssemblyLoadContext context, AssemblyName assemblyName)
+        private Assembly? ResolverForRoslyn(AssemblyLoadContext context, AssemblyName assemblyName)
         {
             return LoadRoslyn(assemblyName, path => context.LoadFromAssemblyPath(path));
         }
 #else
-        private Assembly ResolverForRoslyn(object sender, ResolveEventArgs args)
+        private Assembly? ResolverForRoslyn(object sender, ResolveEventArgs args)
         {
             AssemblyName name = new(args.Name);
             return LoadRoslyn(name, path => Assembly.LoadFrom(path));
         }
 #endif
 
-        private Assembly LoadRoslyn(AssemblyName name, Func<string, Assembly> loadFromPath)
+        private Assembly? LoadRoslyn(AssemblyName name, Func<string, Assembly> loadFromPath)
         {
             const string codeAnalysisName = "Microsoft.CodeAnalysis";
             const string codeAnalysisCsharpName = "Microsoft.CodeAnalysis.CSharp";
             if (name.Name == codeAnalysisName || name.Name == codeAnalysisCsharpName)
             {
-                Assembly asm = loadFromPath(Path.Combine(RoslynAssembliesPath, $"{name.Name}.dll"));
-                Version resolvedVersion = asm.GetName().Version;
+                Assembly asm = loadFromPath(Path.Combine(RoslynAssembliesPath!, $"{name.Name}.dll"));
```

Comment:
This is an input that you are now marking as nullable, so you can't assert it is not null with an exclamation point.

Reply:
Either mark the parameter as non-nullable, or check for null and throw an exception.

Reply:
The parameter is already marked as required, that's how msbuild makes sure that a parameter must be set and isn't null. The parameter itself must be marked as nullable as otherwise the compiler complains that the value isn't set inside the constructor. Nullability doesn't work well with msbuild required properties and using the bang operator here is the best way that I could find to suppress these false positives.